woocommerce/woocommerce-ios
# Dynamic Dashboard: Add new card type and start inbox card UI

Part of #12684 


## Description

This PR adds a few changes to prepare for M2 of Dynamic Dashboard:
- New feature flag.
- New card types.
- Updated all cards to display the card type names to avoid having duplicated localized strings.
- Prep for Inbox card: only the header is available for now.

## Testing instructions

- Enable the feature flag `dynamicDashboardM2` and build the app.
- Confirm that the dashboard still has the same layout as before.
- Tap Edit button and confirm that there's an option for the Inbox card and it's disabled.
- Enable Inbox and save the change.
- Confirm that the Inbox card is displayed on the dashboard.
- Tap the ellipsis menu on the Inbox card, confirm that there's an option Hide Inbox. 
- Tap Hide Inbox, confirm that the customize screen is displayed.

Disable the `dynamicDashboardM2` flag and confirm that the Inbox card is not available.

## Screenshots



Customize | Inbox card
--- | ---
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/8d6d56e0-838c-4a77-bb2a-27912afa6052" width=320 /> |  <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/99976f4c-3eb6-4e9a-b968-a493bb7c9f4c" width=320 />


---
